tracer/src/Datadog.Trace.SourceGenerators/InstrumentationDefinitions/Sources.cs

```diff
@@ -35,6 +35,48 @@ void BuildInstrumentationDefinitions(StringBuilder sb, List<CallTargetDefinition
             }};");
             }
 
+            static void BuildInstrumentedAssemblies(StringBuilder sb, IReadOnlyCollection<CallTargetDefinitionSource> orderedDefinitions)
+            {
+                var hashSet = new HashSet<string>();
+                foreach (var orderedDefinition in orderedDefinitions)
+                {
+                    if (!IsKnownAssemblyPrefix(orderedDefinition.AssemblyName))
+                    {
+                        hashSet.Add(orderedDefinition.AssemblyName);
+                    }
+                }
+
+                sb.Append(
+                    $$"""
+                    #if NETCOREAPP
+                                    new HashSet<string>({{hashSet.Count}}, StringComparer.Ordinal)
+                    #else
+                                    new HashSet<string>(StringComparer.Ordinal)
+                    #endif
+                                    { 
+                    """);
+
+                foreach (var assembly in hashSet.OrderBy(x => x))
+                {
+                    sb.Append('"')
+                      .Append(assembly)
+                      .Append("\", ");
+                }
+
+                sb.AppendLine("};");
+                return;
+
+                // NOTE: Keep this in sync with ExceptionRedactor.IsKnownAssemblyPrefix()
+                static bool IsKnownAssemblyPrefix(string assemblyName)
+                {
+                    return assemblyName.StartsWith("Datadog.", StringComparison.Ordinal)
+                        || assemblyName.StartsWith("mscorlib,", StringComparison.Ordinal)
```

Comment:
Most of these entries end with `.` but `"mscorlib,"` ends with `,`. Intentional?

Reply:
Added a comment, thanks 🙂